asllib/Typing.ml

```diff
@@ -2699,6 +2709,7 @@ module Annotate (C : ANNOTATE_CONFIG) : S = struct
         (* Rule LCFD: A ~local declaration shall not declare an identifier
            which is already in scope at the point of declaration. *)
         let+ () = check_var_not_in_env ~loc env x in
+        let+ () = check_no_precision_loss ~loc ty in
```

Comment:
It feels a bit surprising that this is here, as `ty` is sometimes a declared type and sometimes an inferred type. I wonder if it might be clearer to delete the check here, and insert it in two places:
- the `SDecl.None` case of `annotate_stmt`
- `inherit_integer_constraints`

I believe this also helps line 3783 below (which I think can be deleted)